hazelcast/src/main/java/com/hazelcast/spi/properties/ClusterProperty.java

```diff
@@ -1884,6 +1884,28 @@ private int getWhenNoSSLDetected() {
     public static final HazelcastProperty JAR_UPLOAD_DIR_PATH
             = new HazelcastProperty("hazelcast.cluster.jarupload.dirpath");
 
+    /**
+     * Defines whether WAN replication events should be fired when values are evicted
+     * from {@link IMap} objects.
+     * <p>
+     * The default value is {@code false}.
+     *
+     * @since 5.4
+     */
+    public static final HazelcastProperty WAN_REPLICATE_IMAP_EVICTIONS
```

Comment:
can you please also javaDoc expected use-case here? 

Because as far as i understand from this change, enabling wan replicated evictions is valid under very specific conditions like you want exact copy of source on target and target has no eviction configured. And then if target takes the role of active cluster, you need to enable evictions on this new target not to go OOM, to enable evictions you need MC connected to target first then you can enable eviction by using it(during the time of enabling you can still go OOM) but this breaks the purpose of automatic switch between clusters(Blue/Green) then later if your initial target comes back as active, things may get more complicated and you possibly need to do similar adjustments again. 

Having consequences will be very helpful for a user.

Reply:
Great point @ahmetmircik, thanks for this feedback. I've expanded the java doc in https://github.com/hazelcast/hazelcast/pull/24941/commits/87b7b613d8313916c7270f4a0082bc2df19dc247, let me know if you think anything else is needed 👍